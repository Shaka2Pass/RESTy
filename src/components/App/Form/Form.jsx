import React from 'react';
import PropTypes from 'prop-types';

const Form = ({ onSubmit, inputURL, buttonText = 'Go!', body, handleRadioButton  }) => (
  <section>
    <form onSubmit={onSubmit}>
      <input type ="text" name="URL" value={inputURL}/>
      <button>{buttonText}</button>
      <label> 
        Get: <input type ="radio" name="urlSearch" value="get" onChange={handleRadioButton}></input>
        Post:<input type ="radio" name="urlSearch" value="post" onChange={handleRadioButton}></input>
        Put:<input type ="radio" name="urlSearch" value="put" onChange={handleRadioButton}></input>
        Patch:<input type ="radio" name="urlSearch" value="patch" onChange={handleRadioButton}></input>
        Delete: <input type ="radio" name="urlSearch" value="delete" onChange={handleRadioButton}></input>
      </label>
      <textarea name="RawJsonBody" value={body}></textarea>
    </form>
  </section>
);

Form.propTypes = {
  onSubmit: PropTypes.func.isRequired,
  inputURL: PropTypes.string.isRequired,
  buttonText: PropTypes.string.isRequired,
  body: PropTypes.string.isRequired,
  handleRadioButton: PropTypes.func.isRequired,
};

export default Form;





