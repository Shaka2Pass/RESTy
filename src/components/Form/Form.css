.Request {
  input {
    display: block;
    margin: 1rem 0;
  }
  label {
    padding: 0.3rem 0.4rem;
    margin: 1rem 0.5rem;
    cursor: hand;
  }
  input [type="radio"] {
    padding: 0.3rem 0.5rem;
    display: none;
  }
  input[type="radio"]:checked + label {
    background-color: rgb(111, 155, 75);
  }
  textarea {
    display: block;
    margin: 1rem 0;
  }
  button{
    display: block;
  }
}
