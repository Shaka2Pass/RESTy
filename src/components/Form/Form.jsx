import React from 'react';
import PropTypes from 'prop-types';

const Form = ({ onSubmit, onUrlChange, inputURL, buttonText = 'Go!', body, handleRadioButton, onBodyChange }) => (
  <section>
    <form onSubmit={onSubmit}>
      <input type ="text" name="URL" value={inputURL} onChange={onUrlChange}/>
      <button>{buttonText}</button>
      <label> 
        Get: <input type ="radio" name="method" value="GET" onChange={handleRadioButton} /> 
        Post:<input type ="radio" name="method" value="POST" onChange={handleRadioButton} />
        Put:<input type ="radio" name="method" value="PUT" onChange={handleRadioButton} />
        Patch:<input type ="radio" name="method" value="PATCH" onChange={handleRadioButton} />
        Delete: <input type ="radio" name="method" value="DELETE" onChange={handleRadioButton} />
      </label>
      <textarea name="RawJsonBody" value={body} onChange={onBodyChange}> </textarea>
    </form>
  </section>
);

Form.propTypes = {
  onSubmit: PropTypes.func.isRequired,
  onUrlChange: PropTypes.func.isRequired,
  inputURL: PropTypes.string.isRequired,
  buttonText: PropTypes.string.isRequired,
  body: PropTypes.string.isRequired,
  handleRadioButton: PropTypes.func.isRequired,
  onBodyChange: PropTypes.func.isRequired,
};

export default Form;





