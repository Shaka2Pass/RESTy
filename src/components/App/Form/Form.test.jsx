import React from 'react';
import { shallow } from 'enzyme';
import Form from './Form';

describe('Form container', () => {
  it('matches a snapshot of Form.jsx', () => {
    const wrapper = shallow(<Form
      onSubmit={()=>{}}
      inputURL="https://xfiles-api.herokuapp.com/api/v1/characters"
      buttonText="Go"
      body="JSON body"
    />);
    expect(wrapper).toMatchSnapshot();
  });
});

