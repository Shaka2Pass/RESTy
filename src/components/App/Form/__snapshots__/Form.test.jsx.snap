// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form container matches a snapshot of Form.jsx 1`] = `
<section>
  <form
    onSubmit={[Function]}
  >
    <input
      name="URL"
      type="text"
      value="https://xfiles-api.herokuapp.com/api/v1/characters"
    />
    <button>
      Go
    </button>
    <label>
      Get: 
      <input
        name="urlSearch"
        type="radio"
        value="get"
      />
      Post:
      <input
        name="urlSearch"
        type="radio"
        value="post"
      />
      Put:
      <input
        name="urlSearch"
        type="radio"
        value="put"
      />
      Patch:
      <input
        name="urlSearch"
        type="radio"
        value="patch"
      />
      Delete: 
      <input
        name="urlSearch"
        type="radio"
        value="delete"
      />
    </label>
    <textarea
      name="RawJsonBody"
      value="JSON body"
    />
  </form>
</section>
`;
